### PR TITLE
Update arrow/parquet to arrow/parquet `53.3.0`

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

- Related to https://github.com/apache/arrow-rs/issues/6597

## Rationale for this change

There are some other fixes waiting on this arrow release so let's get them integrated into DataFusion

## What changes are included in this PR?

Update to latest arrow/parquet

## Are these changes tested?
By CI

### CODE REVIEW

**Comment by jayzhan211** [comment]
Why is CI forced to run in arrow 53.3 🤔 ?

**Comment by jayzhan211** [comment]
@findepi It seems https://github.com/apache/datafusion/issues/13304 is not yet resolved but we have related test for preventing regression. Unfortunately, this is also blocking the CI to pass.

I think we should either find a way to keep CI run in 53.2 version or remove the related test.

**Comment by jayzhan211** [comment]
I push the code since this is blocking CI.

**Review by jonahgao — APPROVED** [review]

**Comment by jonahgao** [comment]
Thanks @alamb @jayzhan211

**Comment by jonahgao** [comment]
> Why is CI forced to run in arrow 53.3 🤔 ?

It seems that CI uses the latest version of dependencies that are semver compatible

**Comment by alamb** [comment]
> Why is CI forced to run in arrow 53.3 🤔 ?

My thinking was that explicitly using arrow 53.3 will ensure that any bugs we fix that rely on arrow 53.3 features will work correctly.

If we don't update DataFusion to explicitly use 53.3.0 in Cargo.toml, CI will run (and use 53.3.0) but other projects with DataFusion could potentially use 53.2.0. If we add code to DataFusion that relies on 53.3.0 (like some of the `LIKE` fixes) this would be broken

File: datafusion/sqllogictest/test_files/string/string_view.slt [hunk 1]
```diff
@@ -39,8 +39,19 @@ drop table test_source
 
 # TODO: Revisit this issue after upgrading to the arrow-rs version that includes apache/arrow-rs#6671.
```
Comment (findepi) [inline datafusion/sqllogictest/test_files/string/string_view.slt:40]: This became now addressable - https://github.com/apache/datafusion/pull/13538

**Pull request merged**